src/twitter/twitter-notifications/twitternotificationsyncadaptor.cpp

```diff
@@ -0,0 +1,628 @@
+    if (ok) {
+        if (!tweets.size()) {
+            SOCIALD_LOG_DEBUG("no mentions received for account" << accountId);
+            decrementSemaphore(accountId);
+            return;
+        }
+
+        int mentionsCount = 0;
+        QString body;
+        QString summary;
+        QDateTime timestamp;
+        QString link;
+        foreach (const QJsonValue &tweetValue, tweets) {
+            QJsonObject tweet = tweetValue.toObject();
+            QDateTime createdTime = parseTwitterDateTime(tweet.value(QLatin1String("created_at")).toString());
+            QString mentionId = tweet.value(QLatin1String("id_str")).toString();
+            QString text = tweet.value(QLatin1String("text")).toString();
+            QJsonObject user = tweet.value(QLatin1String("user")).toObject();
+            QString userName = user.value(QLatin1String("name")).toString();
+            QString userScreenName = user.value(QLatin1String("screen_name")).toString();
+
+            // check to see if we need to post it to the notifications feed
+            int sinceSpan = m_accountSyncProfile
+                          ? m_accountSyncProfile->key(Buteo::KEY_SYNC_SINCE_DAYS_PAST, QStringLiteral("7")).toInt()
+                          : 7;
+            if (m_lastSyncTimestamp.isValid() && createdTime < m_lastSyncTimestamp) {
+                SOCIALD_LOG_DEBUG("mention notification for account" << accountId << "came after last sync:" << createdTime << ":" << text);
+                break; // all subsequent notifications will be even older.
+            } else if (createdTime.daysTo(QDateTime::currentDateTimeUtc()) > sinceSpan) {
+                SOCIALD_LOG_DEBUG("mention for account" << accountId << "is more than" << sinceSpan << "days old:" << createdTime << ":" << text);
+            } else {
+                summary = userName;
+                //% "mentioned you in a tweet"
+                body = qtTrId("qtn_social_notifications_twitter_mentioned_you");
+                timestamp = createdTime;
+                link = QLatin1String("https://mobile.twitter.com/") + userScreenName + QLatin1String("/status/") + mentionId;
+                mentionsCount ++;
+            }
+        }
+
+        if (mentionsCount > 0) {
+            // Search if we already have a notification
+            Notification *notification = createNotification(accountId, Mention);
+
+            // Set properties of the notification
+            notification->setItemCount(notification->itemCount() + mentionsCount);
+            QStringList openUrlArgs;
+            if (notification->itemCount() == 1) {
+                notification->setTimestamp(timestamp);
+                notification->setSummary(summary);
+                notification->setBody(body);
+                openUrlArgs << link;
+            } else {
+                notification->setTimestamp(QDateTime::currentDateTimeUtc());
+                notification->setSummary(qtTrId("qtn_social_notifications_twitter_you"));
```

Comment:
Needs engineering english or a comment saying that the EE text is defined elsewhere.